core/src/main/scala/org/apache/spark/sql/SnappyParser.scala

```diff
@@ -699,6 +702,16 @@ class SnappyParser(session: SnappySession)
     )
   }
 
+  protected final def relationPut: Rule1[LogicalPlan] = rule {
+    relationFactor ~ (
+        join. + ~> ((r1: LogicalPlan, joins: Any) => joins.asInstanceOf[
```

Comment:
In fact I think the join portion should be removed. A putInto will always be onto a single relation and doesn't make sense to have a join here (the source Dataset being put can have joins of course).